src/tox/config/__init__.py

```diff
@@ -446,7 +446,7 @@ def tox_addoption(parser):
         "--showconfig",
         action="store_true",
         help="show live configuration (by default all env, with -l only default targets,"
-        " specific via TOXENV/-e)",
+        " specific via TOXENV/-e). Add -q if you plan to parse it.",
```

Comment:
I wouldn't add that here, as it is also not mentioned in help of -a and -l. It would be nice though, if we had a dedicated little bit in the docs, which explains where tox actually makes a certain promise to produce machine readable output that doesn't change in format. AFAICT this is the case for -l and --showconfig (and maybe for -a for advanced tools?). So this could go into the docs for ... I don't know ... tool builders? 

Do you think you could find a suitable place in the docs and write a little bit about that "promise"? And there it would then need to be explained that `-q` always has to be used in to be sure that provisioning output is not polluting the parseable output.

It definitely feels a bit dirty even writing that, as the question pops up again, why any additional output is produced on stdout for commands that produce machine readable output, but here we are and tox3 is a classic case of "grown organically" and everything will be super consistent and correct in tox4 :)